src/Common/FileSegment.h

```diff
@@ -144,9 +135,11 @@ friend class FileSegmentRangeWriter;
 
     size_t getDownloadedSize() const;
 
+    size_t getRemainingSizeToDownload() const;
+
     void completeBatchAndResetDownloader();
 
-    void complete(State state);
+    void complete(State state, bool auto_resize = false);
```

Comment:
I do not want to get rid of default argument here, because the case when this argument changes to non-default is a very specific case and should not be used anywhere apart from this case.